mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/events/ElasticSearchListeningMessageSearchIndexTest.java

```diff
@@ -103,168 +182,234 @@ public void deserializeElasticSearchListeningMessageSearchIndexGroup() throws Ex
+        testee.delete(session, mailbox, Lists.newArrayList(MESSAGE_UID_1));
+        elasticSearch.awaitForElasticSearch();
+
+        SearchQuery query = new SearchQuery(SearchQuery.all());
+        assertThat(testee.search(session, mailbox, query))
+            .isEmpty();
     }
 
     @Test
-    @SuppressWarnings("unchecked")
-    public void deleteShouldPropagateExceptionWhenExceptionOccurs() throws IOException {
-        //Given
-        when(elasticSearchIndexer.delete(any(List.class)))
-            .thenThrow(new ElasticsearchException(""));
-
-        // Then
-        assertThatThrownBy(() -> testee.delete(session, mailbox, Lists.newArrayList(MESSAGE_UID)))
-            .isInstanceOf(ElasticsearchException.class);
+    public void deleteShouldPropagateExceptionWhenExceptionOccurs() throws Exception {
+        elasticSearch.getDockerElasticSearch().pause();
+        Thread.sleep(Duration.FIVE_SECONDS.getValueInMS()); // Docker pause is asynchronous and we found no way to poll for it
+
+        assertThatThrownBy(() -> testee.delete(session, mailbox, Lists.newArrayList(MESSAGE_UID_1)))
+            .isInstanceOf(IOException.class);
+
+        elasticSearch.getDockerElasticSearch().unpause();
     }
 
     @Test
-    public void updateShouldWork() throws Exception {
-        //Given
-        Flags flags = new Flags();
+    public void updateShouldUpdateIndex() throws Exception {
+        testee.add(session, mailbox, MESSAGE_1);
+        elasticSearch.awaitForElasticSearch();
 
+        Flags newFlags = new Flags(Flags.Flag.ANSWERED);
         UpdatedFlags updatedFlags = UpdatedFlags.builder()
-            .uid(MESSAGE_UID)
-            .modSeq(MODSEQ)
-            .oldFlags(flags)
-            .newFlags(flags)
+            .uid(MESSAGE_UID_1)
+            .modSeq(MOD_SEQ)
+            .oldFlags(new Flags())
+            .newFlags(newFlags)
             .build();
 
-        when(messageToElasticSearchJson.getUpdatedJsonMessagePart(any(Flags.class), any(Long.class)))
-            .thenReturn("json updated content");
-        
-        //When
         testee.update(session, mailbox, Lists.newArrayList(updatedFlags));
-        
-        //Then
-        ImmutableList<UpdatedRepresentation> expectedUpdatedRepresentations = ImmutableList.of(new UpdatedRepresentation(ELASTIC_SEARCH_ID, "json updated content"));
-        verify(elasticSearchIndexer).update(expectedUpdatedRepresentations);
+        elasticSearch.awaitForElasticSearch();
+
+        SearchQuery query = new SearchQuery(SearchQuery.flagIsSet(Flags.Flag.ANSWERED));
+        assertThat(testee.search(session, mailbox, query))
+            .containsExactly(MESSAGE_1.getUid());
     }
 
     @Test
-    public void updateShouldPropagateExceptionWhenExceptionOccurs() throws Exception {
-        //Given
-        Flags flags = new Flags();
+    public void updateShouldNotUpdateNorThrowOnUnknownMessageUid() throws Exception {
```

Comment:
why?

Reply:
and what about delete then?

Reply:
That's the behavior of update in the code. We first search if the message exists with its UID, if it does we update, if it doesn't we do nothing. Thus no exception.

I thought that `deleteShouldBeIdempotent` was proving implicitly that case but I can add an extra test just for this yes !